From f56249dd6e79cf781da4191421bea477dc1fe5bc Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Thu, 6 Oct 2022 19:05:19 +0100
Subject: [PATCH] GBMUtils: LockFrontBuffer should return the front of queue

We are using emplace to add to back of queue, and pop to remove from front.
We should be returning the buffer from front of queue, not the one just added.

I believe we are currently only using 2 buffers, when mesa makes 4 available
---
 xbmc/windowing/gbm/GBMUtils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/windowing/gbm/GBMUtils.cpp b/xbmc/windowing/gbm/GBMUtils.cpp
index 5267c93c8f3d7..15c4cc9d1cd0b 100644
--- a/xbmc/windowing/gbm/GBMUtils.cpp
+++ b/xbmc/windowing/gbm/GBMUtils.cpp
@@ -92,7 +92,7 @@ CGBMUtils::CGBMDevice::CGBMSurface::CGBMSurfaceBuffer* CGBMUtils::CGBMDevice::CG
     m_buffers.pop();
   }
 
-  return *m_buffers.back();
+  return *m_buffers.front();
 }
 
 CGBMUtils::CGBMDevice::CGBMSurface::CGBMSurfaceBuffer::CGBMSurfaceBuffer(gbm_surface* surface)
